src/consumer.coffee

```diff
@@ -1,7 +1,52 @@
+crypto            = require 'crypto'
+URL         = require 'url-parse'
+HMAC_SHA1         = require './hmac-sha1'
+errors            = require './errors'
+extensions        = require './extensions'
+
 class Consumer
-  constructor: () ->
-    console.error('Consumer Not Implemented')
-    false
+  constructor: (consumer_key, consumer_secret) ->
+    if typeof consumer_key is 'undefined' or consumer_key is null
+      throw new errors.ConsumerError 'Must specify consumer_key'
+
+    if typeof consumer_secret is 'undefined' or consumer_secret is null
+      throw new errors.ConsumerError 'Must specify consumer_secret'
+
+    @consumer_key     = consumer_key
+    @consumer_secret  = consumer_secret
+    @signer           = new HMAC_SHA1()
+    @body             = {}
+
+  sign_request: (url, method = 'GET', body = {}) =>
```

Comment:
All of those details are around proper oauth signatures, which an oauth signing library would also take care of.

Is there no lti-specific details that one would also want to take care of here?